### DISCUSSION TITLE
internal links in Canvas notes

### DISCUSSION THREAD

**Discussion by cbagdatli** [TriliumNext/Trilium#4464] [General]
Hello I couldnt discover a way to add internal notes links to objects in Canvas notes.
Ctrl L is not working and I didnt see in the menus.
How can we add internal links to objects in canvas? using 0.62.2

**Comment by rodrigoalexandrecosta** [comment]
Internal links are links in the following format (for instance) `#root/xE8a12eR`, where `#root` is always present since it is, well, you know, the root, and each forward slash represents a step down in the tree structure/hierarchy, which means that an internal link is basically the note path taking the `noteId`s into consideration.

You now can obtain the note path right-clicking in the note you want to link in the tree -> Advanced -> Copy note path to clipboard, and then you can paste that link in any canvas shape. Then, when you click in the link icon at the top right of that shape, it will direct you to the note.

**Reply by cbagdatli** [reply]
thank you :)
